ml-algorithms/src/test/java/org/opensearch/ml/engine/algorithms/remote/AwsConnectorExecutorTest.java

```diff
@@ -129,9 +127,9 @@ public void executePredict_RemoteInferenceInput_NullResponse() throws IOExceptio
             .url("http://test.com/mock")
             .requestBody("{\"input\": \"${parameters.input}\"}")
             .build();
-        Map<String, String> credential = ImmutableMap
+        Map<String, String> credential = Map
             .of(ACCESS_KEY_FIELD, encryptor.encrypt("test_key"), SECRET_KEY_FIELD, encryptor.encrypt("test_secret_key"));
-        Map<String, String> parameters = ImmutableMap.of(REGION_FIELD, "us-west-2", SERVICE_NAME_FIELD, "sagemaker");
+        Map<String, String> parameters = Map.of(REGION_FIELD, "us-west-2", SERVICE_NAME_FIELD, "sagemaker");
```

Comment:
Can we keep this ImmutableMap at least for these ones? It doesn't look like we want to update them in the code. We can use the equivalent in aws sdk through "import software.amazon.awssdk.utils.ImmutableMap"

Reply:
Good question. The Map.of() only gives you the very basic immutableMap which is not convenient to handle Null value, factory pattern, builder pattern etc. I think you will find more advantages by searching the difference online. because AWS already provided the same ImmutableMap, why not use it without adding any new dependencies?

Reply:
I think it's better to use core language unless there is a clear reason to add new external dependency with a clear use case to show why those are needed. So far I haven't seen one and honestly the shaded repackage thing breaks my IDE (besides mess of exclusions in the gradle.build), which is kinda annoying as well.

Specifically regarding the AWS SDK vs Guava, I would rather keep Guava rather than rely on AWS SDK implementation.

With that being said, if the community doesn't want to change this dependency I'm fine abandoning this as well.

Reply:
> shaded repackage thing breaks my IDE

try to using latest IntelliJ IDEA version of 2023.3.2, the shadow jar was fix in this version.